File: tests/objectives/optimization_class_test.cpp
//=============================================================================
//    Copyright (C) 2021-2022 Wageningen University - All Rights Reserved
//                     Author: Gonzalo Mier
//                        BSD-3 License
//=============================================================================

#include <gtest/gtest.h>
#include "fields2cover/objectives/optimization_class.h"
#include "fields2cover/objectives/n_swath.h"
#include "fields2cover/types.h"

TEST(fields2cover_obj_optimization_class, computeCost_cost) { 
  F2CSwaths swaths50 {F2CSwaths(50)};

  f2c::obj::OptimizationClass<f2c::obj::NSwath> opt;

  EXPECT_EQ(opt.computeCost(swaths50), 50.0);
  swaths50.emplace_back(F2CSwath());
  EXPECT_EQ(opt.computeCost(swaths50), 51.0);
}

File: tests/python/objectives/optimization_class_test.py
#==============================================================================
#     Copyright (C) 2021-2022 Wageningen University - All Rights Reserved
#                      Author: Gonzalo Mier
#                         BSD-3 License
#==============================================================================

import pytest
import fields2cover as f2c

def near(a, b):
  assert pytest.approx(a) == pytest.approx(b)


def test_fields2cover_obj_optimization_class_computeCost():
  swaths50 = f2c.Swaths(50);

  opt = f2c.OBJ_Optimization_NSwath();

  near(opt.computeCost(swaths50), 50.0);
  swaths50.push_back(f2c.Swath());
  near(opt.computeCost(swaths50), 51.0);
